src/raftstore/store/peer.rs

```diff
@@ -408,6 +408,14 @@ impl Peer {
         let mut ready = self.raft_group.ready();
         let is_applying = self.get_store().is_applying_snap();
         if is_applying {
+            if !raft::is_empty_snap(&ready.snapshot) {
+                if !self.get_store().is_canceling_snap() {
+                    warn!("receiving a new snap {:?} when applying the old one, try to abort.",
+                          ready.snapshot);
+                    self.mut_store().cancle_applying_snap();
```

Comment:
No, the same snapshot won't be sent again once it's applied in the state machine.